src/github.com/matrix-org/go-neb/services/imgur/imgur.go

```diff
@@ -0,0 +1,268 @@
+	// vote	string	The current user's vote on the album. null if not signed in or if the user hasn't voted on it.
+	// favorite	boolean	Indicates if the current user favorited the image. Defaults to false if not signed in.
+	// account_url	string	The username of the account that uploaded it, or null.
+	// account_id	integer	The account ID of the account that uploaded it, or null.
+	// ups	integer	Upvotes for the image
+	// downs	integer	Number of downvotes for the image
+	// points	integer	Upvotes minus downvotes
+	// score	integer	Imgur popularity score
+}
+
+type imgurGalleryAlbum struct {
+	ID          string              `json:"id"`           // The ID for the album
+	Title       string              `json:"title"`        // The title of the album.
+	Description string              `json:"description"`  // Description of the album.
+	DateTime    int64               `json:"datetime"`     // Time inserted into the gallery, epoch time
+	Views       int64               `json:"views"`        // The number of album views
+	Link        string              `json:"link"`         // The URL link to the album
+	NSFW        bool                `json:"nsfw"`         // Indicates if the album has been marked as nsfw or not. Defaults to null if information is not available.
+	Topic       string              `json:"topic"`        // Topic of the gallery album.
+	IsAlbum     bool                `json:"is_album"`     // If it's an album or not
+	Cover       string              `json:"cover"`        // The ID of the album cover image
+	CoverWidth  int                 `json:"cover_width"`  // The width, in pixels, of the album cover image
+	CoverHeight int                 `json:"cover_height"` // The height, in pixels, of the album cover image
+	ImagesCount int                 `json:"images_count"` // The total number of images in the album
+	Images      []imgurGalleryImage `json:"images"`       // An array of all the images in the album (only available when requesting the direct album)
+
+	// ** Unimplemented fields **
+	// account_url	string	The account username or null if it's anonymous.
+	// account_id	integer	The account ID of the account that uploaded it, or null.
+	// privacy	string	The privacy level of the album, you can only view public if not logged in as album owner
+	// layout	string	The view layout of the album.
+	// views	integer	The number of image views
+	// ups	integer	Upvotes for the image
+	// downs	integer	Number of downvotes for the image
+	// points	integer	Upvotes minus downvotes
+	// score	integer	Imgur popularity score
+	// vote	string	The current user's vote on the album. null if not signed in or if the user hasn't voted on it.
+	// favorite	boolean	Indicates if the current user favorited the album. Defaults to false if not signed in.
+	// comment_count	int	Number of comments on the gallery album.
+	// topic_id	integer	Topic ID of the gallery album.
+}
+
+type imgurSearchResponse struct {
+	Data    []json.RawMessage `json:"data"`
```

Comment:
I've added a comment to explain that the response data is a mix of imgurGalleryImage and imgurGalleryAlbum objects (to be parsed at a later date). Hopefully this is what you were thinking of?